### ISSUE TITLE
reverseproxy subfolder

### ISSUE THREAD

**Issue by GettingTechnicl** [denverquane/automuteus#195]
Hello, 

I am using Nginx in a docker, I also have the amongusbot hosted in a docker. The amongusbot is fully accessible and reachable via port, although I would like to configure this to work with proxy so I can close the port ofc. 

the specific nginx docker that I'm using is "linuxserver/swag" 

This docker has many pre-built templates for subdomain and subfolder for various apps that are commonly used, In creating this I decided I would copy the mylar subfolder template. 

the docker running the discord bot is named "dbot" and port is set to "8123"
The nginx docker and discordbot are both set to run on the same bridge network that I have created. 

After editing the template for mylar proxy through subfolder it looks like this. 

![image](https://user-images.githubusercontent.com/56567784/97375425-c6829e80-1888-11eb-9843-18176c3ff436.png)

I have also tried ## out the proxy_pass line that's there and creating a new one using the docker ip, both of these give me same result, when trying to connect with the Among Us Capture Client I get an error code 200. 

Looking for some assistance to get this set up and working, thank you.

**Comment by ShawnHardwick** [comment]
Really don't think this belongs in this project. It's not anything unique to the bot.

**Comment by GettingTechnicl** [comment]
I would say this is specific to the bot. 

For HOST= I can't put in a domain name with a /subfolder as the bot errors and see's a problem with it. 

I have multiple reverse proxy's working on other machines without issues, I am unable to connect to this bot, because either the way the capture client is set to connect, or because of how the bot is set to receive connections. I am doing nothing more than trying to point out the issues that are preventing me from connecting to this specific bot. 

I am quite sure my current proxy setup is correct, as the bot is unable to set a / base url 

This would not go in an nginx forum, nginx works fine and can forward subfolders fine

This would not go in to the docker forum, as this docker is doing exactly what it's supposed to be doing and I'm using it with other applications on other servers. 

So this should go here because this issue is specific to this application. If you have a better idea of where this "should go" please inform me. 

![image](https://user-images.githubusercontent.com/56567784/97480540-8fac9700-1921-11eb-8f56-d378dc0d4fac.png)

**Comment by denverquane** [comment]
Yeah paths like `https://reverse-proxy.com:443/path` are not supported. Reverse proxy support right now is limited to redirecting from the base `/` path on a different port, but not subdomain paths. Happy to accept a PR that introduces this (it would be a reconfiguration of the regex specified here: https://github.com/denverquane/automuteus/blob/master/discord/message_handlers.go#L34), but it's not a priority of mine, personally.

**Comment by INGY23** [comment]
Had a quick play and couldn't get it to work unfortunately.

Since this uses web sockets you will also probably need the following if you do find a way to make it work

proxy_set_header Upgrade $http_upgrade;
proxy_set_header Connection "upgrade";
proxy_hide_header X-Frame-Options;
